src/server/data-access/MwBotWikibaseMessagesRepo.ts

```diff
@@ -0,0 +1,75 @@
+import TechnicalProblem from '@/common/data-access/error/TechnicalProblem';
+import MessageNotFound from '@/common/data-access/error/MessageNotFound';
+import MessageTranslationCollection from '@/datamodel/MessageTranslationCollection';
+import MessageCollection from '@/datamodel/MessageCollection';
+import MessagesRepository from '@/common/data-access/MessagesRepository';
+import mwbot from 'mwbot';
+
+interface RespondMessage {
+	[ key: string]: string;
+}
+
+export default class MwBotWikibaseMessagesRepo implements MessagesRepository {
+	private bot: mwbot;
+	private messageKeys: string[];
+
+	public constructor( bot: mwbot, messageKeys: string[] ) {
+		this.bot = bot;
+		this.messageKeys = messageKeys;
+	}
+
+	public getMessagesInLanguage( inLanguage: string ): Promise<MessageTranslationCollection> {
+		return new Promise( ( resolve, reject ) => {
+			this.bot.request( {
+				action: 'query',
+				meta: 'allmessages',
+				ammessages: this.messageKeys.join( '|' ),
+				amlang: inLanguage,
+			} )
+				.then( ( response: any ) => {
+					if ( !( 'query' in response ) ) {
+						reject( new TechnicalProblem( 'allmessages result not well formed.' ) );
+						return;
+					}
+
+					if ( !( 'allmessages' in response.query ) ) {
+						reject( new TechnicalProblem( 'allmessages result not well formed.' ) );
+						return;
+					}
+
+					let resolved = {};
+					try {
+						resolved = this.getMessageTranslationCollection( inLanguage, response.query.allmessages );
+					} catch ( error ) {
+						reject( error );
+						return;
+					}
+					resolve( resolved );
+				} )
+				.catch( ( reason: string ) => {
+					reject( new TechnicalProblem( reason ) );
+				} );
+		} );
+	}
+
+	private getMessageTranslationCollection(
+		inLanguage: string,
+		response: RespondMessage[],
+	): MessageTranslationCollection {
+		const messageCollection: MessageTranslationCollection = {};
+		messageCollection[ inLanguage ] = this.transformMessages( response );
+		return messageCollection;
+	}
+
+	private transformMessages( response: RespondMessage[] ): MessageCollection {
+		const messageList: MessageCollection = {};
+		response.forEach( ( value: { [key: string]: string } ) => {
```

Comment:
`value` here is of type `RespondMessage`, so the hint and the name could be better

Reply:
done